### ISSUE TITLE
Adding other metrics for validation steps

### ISSUE THREAD

**Issue by marcmk6** [songlab-cal/tape#125]
Hi,

I'm doing classification of sequences with `ProteinBertForSequenceClassification`. I'm wondering what's an appropriate way to add some metrics (e.g. ROCAUC, Average Precision) to be shown during the validation steps.

Thanks

**Comment by rmrao** [comment]
The code assumes that the first item returned by the model will be a tuple of (loss, metrics). For sequence classification, the current loss + metrics are calculated and returned by the [SequenceClassificationHead](https://github.com/songlab-cal/tape/blob/6d345c2b2bbf52cd32cf179325c222afd92aec7e/tape/models/modeling_utils.py#L797-L813). In [line 809](https://github.com/songlab-cal/tape/blob/6d345c2b2bbf52cd32cf179325c222afd92aec7e/tape/models/modeling_utils.py#L809), the metrics dict is computed with a single metric, accuracy. You can add arbitrary metrics to this dict, and they will be reduced and shown.

That being said, this code is also not being actively maintained and was created prior to the existence of common training frameworks such as PytorchLightning, fairseq, etc. If you are planning to work extensively with training these models and returning custom metrics, it may be worth investing the time to create a training loop in one of those frameworks, since they will have better support for customization.

**Comment by marcmk6** [comment]
I see. Thanks for your explanation!
